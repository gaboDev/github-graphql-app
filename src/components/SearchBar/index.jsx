import React from "react";
import { string, func } from "prop-types";
import { Paper, TextField } from "@mui/material";
import styled from "@emotion/styled";

const SearchBarContainer = styled(Paper)`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 1em 0;
`;

const Logo = styled.img`
  width: 55px;
  margin-right: 1em;
`;

const Input = styled(TextField)`
  width: 250px;
  border: 0;
  
  &:focus{
    outline: none;
    box-shadow: 0 3px #b1b1b1;
  }
`;

const SearchBar = ({logo, onSearch, onChange, searchValue}) => {
    const handleOnSubmit = (event) => {
        event.preventDefault();
        onSearch(searchValue);
    }
    return (
      <SearchBarContainer square>
          <Logo src={logo} />
          <form onSubmit={handleOnSubmit}>
              <Input label='Search related topics'
                     variant="filled"
                     value={searchValue}
                     name='searchValue'
                     onChange={onChange}
              />
          </form>
      </SearchBarContainer>
    );
}

SearchBar.defaultProps = {
  searchValue: null
}

SearchBar.propTypes = {
  logo: string.isRequired,
  onSearch: func.isRequired,
  onChange: func.isRequired,
  searchValue: string
};

export default SearchBar;
