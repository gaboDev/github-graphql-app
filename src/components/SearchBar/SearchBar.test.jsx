import { render, fireEvent, screen } from '@testing-library/react';
import SearchBar from "./index";


describe('<SearchBar />', () => {
    const initialProps = {
        logo: 'img/github.png',
        onSearch: () => {},
        onChange: () => {},
        searchValue: 'reactjs'
    };
    const renderSearchBar = ({ logo, onSearch, onChange, searchValue } = initialProps) => (
        render(
            <SearchBar
                onChange={onChange}
                onSearch={onSearch}
                logo={logo}
                searchValue={searchValue}
            />
        )
    );
    it('should match snapshot', () => {
        const view = renderSearchBar();
        expect( view ).toMatchSnapshot();
    });
    it('should display an image', () => {
        renderSearchBar();
        const logo = screen.getByRole('img');
        expect( logo ).toBeInTheDocument();
    });
    it('should handle onChange event', () => {
        const onChange = jest.fn();
        renderSearchBar({ ...initialProps, onChange });
        const input = screen.getByRole("textbox");
        fireEvent.change(input, {target: { value: "Hello!" }});
        expect(onChange).toHaveBeenCalled();
    });
    it('should handle onSearch properly', () => {
        const searchValue = "reactjs";
        const onSearch = jest.fn((topic) => {
            expect(topic === searchValue).toBeTruthy();
        });
        renderSearchBar({ ...initialProps, onSearch, searchValue });
        const input = screen.getByRole("textbox");
        fireEvent.submit(input);
        expect(onSearch).toHaveBeenCalled();
    });
});