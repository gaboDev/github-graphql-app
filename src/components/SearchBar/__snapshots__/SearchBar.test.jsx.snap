// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 css-y3c26e-MuiPaper-root"
      >
        <img
          class="css-fx74e"
          src="img/github.png"
        />
        <form>
          <div
            class="MuiFormControl-root MuiTextField-root css-qrg1pc-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r0:"
              id=":r0:-label"
            >
              Search related topics
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
                id=":r0:"
                name="searchValue"
                type="text"
                value="reactjs"
              />
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 css-y3c26e-MuiPaper-root"
    >
      <img
        class="css-fx74e"
        src="img/github.png"
      />
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root css-qrg1pc-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            Search related topics
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              id=":r0:"
              name="searchValue"
              type="text"
              value="reactjs"
            />
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
